src/@types/options.ts

```diff
@@ -0,0 +1,93 @@
+import type { PartialDeep } from 'type-fest';
+import { ILevel, ILogger } from './logger';
+import { ISerializer } from './serializer';
+
+/**
+ * A function that takes in the `PrismaSessionStore` options, the `SessionData`
+ * and the `sid` for a given session and returns the TTL (time to live) in milliseconds
+ * for that particular session.
+ */
+export type TTLFactory = (
```

Comment:
😅  Not exactly sure myself this is carry over from `memorystore` which I left in to avoid any breaking changes with this rewrite.

You will notice that, other then type information, there are no breaking changes in this release (as far as I can tell).
I suppose we could add better documentation for this but that should be done at the `IOptions` interface so that it is exposed to the user rather then on an internal type.

Reply:
Got it. Perhaps in `memorystore` it was anticipated that this would be useful for testing? Perhaps it _is_ useful in ways I haven't seen? It might be [cargo culting](https://en.wikipedia.org/wiki/Cargo_cult_programming), but I'm leaning towards not disturbing the way things have been, here (assuming you don't have strong feelings in the other direction)...